src/language.rs

```diff
@@ -22,4 +51,40 @@ impl Language {
             )),
         }
     }
+
+    pub fn detect(filename: &str) -> FormatterResult<&str> {
+        // NOTE This extension search is influenced by Wilfred Hughes' Difftastic
+        // https://github.com/Wilfred/difftastic/blob/master/src/parse/guess_language.rs
+        if let Some(extension) = Path::new(filename).extension() {
+            let extension = extension.to_str().unwrap();
+
+            for (language, extensions) in EXTENSIONS {
+                for candidate in extensions.iter() {
```

Comment:
Here you can use https://doc.rust-lang.org/rust-by-example/fn/closures/closure_examples/iter_any.html

Reply:
Changed in 40c4d5d467d7d36b8